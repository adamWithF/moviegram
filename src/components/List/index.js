import React from "react"
import propTypes from "prop-types"

const List = ({ list, title, className, Item }) => (
  <section className={className}>
    {title && <header>{title}</header>}
    <ul>
      {list.map((s, i) => (
        <li key={i}>{Item ? <Item {...s} /> : s}</li>
      ))}
    </ul>
    {!list.length && <span className="no-records">no records</span>}
  </section>
)

List.propTypes = {
  list: propTypes.array.isRequired,
  title: propTypes.string,
  Item: propTypes.func,
  className: propTypes.string,
  itemName: propTypes.string
}

List.defaultProps = {
  className: "",
  itemName: "record"
}

export default List
