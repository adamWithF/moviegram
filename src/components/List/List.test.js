import React from "react"
import { shallow } from "enzyme"

import List from "../List"

describe("List", () => {
  let props

  beforeEach(() => {
    props = {
      list: []
    }
  })

  it("should render properly when empty list provided", () => {
    const wrapper = shallow(<List {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it("should render properly when no empty list provided", () => {
    props.list = [1, 2]
    const wrapper = shallow(<List {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
