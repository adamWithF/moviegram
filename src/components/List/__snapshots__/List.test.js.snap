// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render properly when empty list provided 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    className=""
    itemName="record"
    list={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <ul>
          
</ul>,
        <span
          className="no-records"
>
          no records
</span>,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "no records",
          "className": "no-records",
        },
        "ref": null,
        "rendered": "no records",
        "type": "span",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <ul>
            
</ul>,
          <span
            className="no-records"
>
            no records
</span>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "no records",
            "className": "no-records",
          },
          "ref": null,
          "rendered": "no records",
          "type": "span",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`List should render properly when no empty list provided 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    className=""
    itemName="record"
    list={
        Array [
            1,
            2,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <ul>
          <li>
                    1
          </li>
          <li>
                    2
          </li>
</ul>,
        false,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              1
</li>,
            <li>
              2
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": 2,
            },
            "ref": null,
            "rendered": 2,
            "type": "li",
          },
        ],
        "type": "ul",
      },
      false,
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <ul>
            <li>
                        1
            </li>
            <li>
                        2
            </li>
</ul>,
          false,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                1
</li>,
              <li>
                2
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": 2,
              },
              "ref": null,
              "rendered": 2,
              "type": "li",
            },
          ],
          "type": "ul",
        },
        false,
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
